File: year2/systems-design-implementation/Lab4/Lab4Sharp/Srv/Service.cs
using Lab2Sharp.Domain;
using Lab2Sharp.Repo.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab2Sharp.Srv
{
    public class Service
    {
        private ParticipantRepository participantRepository;
        private ProbaRepository probaRepository;
        private InscriereRepository inscriereRepository;
        private UserRepository userRepository;

        public Service(ParticipantRepository repo1, ProbaRepository repo2, InscriereRepository repo3, UserRepository repo4)
        {
            participantRepository = repo1;
            probaRepository = repo2;
            inscriereRepository = repo3;
            userRepository = repo4;
        }

        public IEnumerable<User> FindAllUseri()
        {
            return userRepository.FindAll();
        }

        public IEnumerable<Proba> FindAllProbe()
        {
            return probaRepository.FindAll();
        }

        public Proba FindProba(int id)
        {
            return probaRepository.FindOne(id);
        }

        public Participant AddParticipant(Participant participant)
        {
            return participantRepository.Add(participant);
        }

        public Inscriere AddInscriere(Inscriere inscriere)
        {
            return inscriereRepository.Add(inscriere);
        }

        public int NrParticipanti(Proba proba)
        {
            return inscriereRepository.NrParticipanti(proba);
        }

        public IEnumerable<Participant> GetInscrisi(Proba proba)
        {
            return inscriereRepository.GetInscrisi(proba);
        }

        public IEnumerable<Proba> GetInscrieri(Participant participant)
        {
            return inscriereRepository.GetInscrieri(participant);
        }

    }
}

File: year2/systems-design-implementation/Lab4/Lab4/src/main/java/srv/Service.java
package srv;

import domain.Inscriere;
import domain.Participant;
import domain.Proba;
import domain.User;
import repo.intf.InscriereRepository;
import repo.intf.ParticipantRepository;
import repo.intf.ProbaRepository;
import repo.intf.UserRepository;

import java.util.List;

public class Service {

    private ParticipantRepository participantRepository;
    private ProbaRepository probaRepository;
    private InscriereRepository inscriereRepository;
    private UserRepository userRepository;

    public Service(ParticipantRepository participantRepository, ProbaRepository probaRepository, InscriereRepository inscriereRepository, UserRepository userRepository) {
        this.participantRepository = participantRepository;
        this.probaRepository = probaRepository;
        this.inscriereRepository = inscriereRepository;
        this.userRepository = userRepository;
    }

    public List<User> findAllUseri() {
        return userRepository.findAll();
    }

    public List<Proba> findAllProbe() {
        return probaRepository.findAll();
    }

    public Participant addParticipant(Participant participant) throws Exception {
        return participantRepository.add(participant);
    }

    public Inscriere addInscriere(Inscriere inscriere) throws Exception {
        return inscriereRepository.add(inscriere);
    }

    public int nrParticipanti(Proba proba) {
        return inscriereRepository.nrParticipanti(proba);
    }

    public List<Participant> getInscrisi(Proba proba) {
        return inscriereRepository.getInscrisi(proba);
    }

    public List<Proba> getInscrieri(Participant participant) {
        return inscriereRepository.getInscrieri(participant);
    }
}
